.tekton/bonfire-component-tests-pipeline.yaml

```diff
@@ -356,33 +358,67 @@ spec:
               echo "  Username: $DATAROUTER_USERNAME"
               echo "  Artifact directory: $ARTIFACT_DIR"
               
-              # Execute            
-              # Create sanitized command for logging (mask password)
-              DROUTE_CMD_SANITIZED=$(echo "$DROUTE_CMD" | sed 's/--password=[^ ]*/--password=***MASKED***/g')
-              echo "About to execute droute command: $DROUTE_CMD_SANITIZED"
-              # Temporarily disable exit on error to capture output even if command fails
-              set +e
-              OUTPUT=$(eval "$DROUTE_CMD" 2>&1)
-              EXIT_CODE=$?
-              set -e
+              # Retry configuration for intermittent 5xx errors
+              MAX_RETRIES=3
+              RETRY_DELAY=45
+              
+              # Function to execute droute command with retry
+              execute_with_retry() {
+                local attempt=1
+                
+                while [ $attempt -le $MAX_RETRIES ]; do
+                  echo "=== Attempt $attempt of $MAX_RETRIES ==="
+                  
+                  # Temporarily disable exit on error to capture output
+                  set +e
+                  OUTPUT=$(/droute send --url=$(params.datarouter-url) --username=$DATAROUTER_USERNAME --password=$DATAROUTER_PASSWORD --metadata=$METADATA_FILE --results=$ARTIFACT_DIR/$(params.results-pattern) $(if [[ -n "$(params.attachments-dir)" ]]; then echo "--attachments=${ARTIFACT_DIR}/$(params.attachments-dir)"; fi) 2>&1)
+                  EXIT_CODE=$?
+                  set -e
+                  
+                  echo "Command completed with exit code: $EXIT_CODE"
+                  echo "Output: $OUTPUT"
+                  
+                  # Check if successful
+                  if [[ $EXIT_CODE -eq 0 ]]; then
+                    echo "Data Router send operation completed successfully"
```

Comment:
@Sgitario this message is also printed at line 416.

Reply:
Updated.